dns-controller/README.md

```diff
@@ -10,13 +10,12 @@ we just expose it via DNS.
 
 The dns-controller recognizes annotations on nodes.
 
-`dns.alpha.kubernetes.io/external` will set up records for accessing the resource externally
-
-`dns.alpha.kubernetes.io/internal` will set up records for accessing the resource internally
+* `dns.alpha.kubernetes.io/external` will set up records for accessing the resource externally
+* `dns.alpha.kubernetes.io/internal` will set up records for accessing the resource internally
 
 When added on `Service` controllers:
 
-`dns.alpha.kubernetes.io/external` creates a Route53 A record with `public` IPs of all the nodes
-`dns.alpha.kubernetes.io/internal` creates a Route53 A record with `private` IPs of all the nodes
+* `dns.alpha.kubernetes.io/external` creates a Route53 A record with the IPs of the load balancer (type=LoadBalancer) or all the nodes (type=NodePort)
+* `dns.alpha.kubernetes.io/internal` creates a Route53 A record with spec.clusterIP of the service
```

Comment:
I agree with @justinsb here. It makes sense to use internal dns names for cluster IPs.

You can even customize the `cluster.local` part to be anything you want.

Actually, you don't even need to query DNS unless absolutely required - service ClusterIPs (and ports) are available as environment variables in the pods. 

More at - https://kubernetes.io/docs/user-guide/services/#discovering-services